openhtf/plugs/acroname/power_module_plug.py

```diff
@@ -0,0 +1,309 @@
+ """Error connecting power module through EtherStem."""
+
+class ConfigurePowerSupplyError(Exception):
+ """Error configuring power supply."""
+
+class TurnOnPowerSupplyError(Exception):
+ """Error turning on power supply."""
+
+class ChangeVoltageError(Exception):
+ """Error changing voltage."""
+
+class ChangeCurrentLimitError(Exception):
+ """Error changing current limit."""
+
+class TurnOffPowerSupplyError(Exception):
+ """Error turning off power supply."""
+
+class PowerModuleShortCircuitError(Exception):
+ """Short Circuit Found on Power Module."""
+
+class DisconnectPowerModuleError(Exception):
+ """Error disconnecting power module."""
+
+
+def CheckReturnCode(code, action_str):
+  if not code:
+    _LOG.info("Result of %s: %d", action_str, code)
+  else:
+    _LOG.error("""Error connecting to power module through 
+      EtherStem while attempting to %s: %d""", action_str,code)
+    raise ConnectingToPowerModuleThroughEtherStemError() 
+
+
+class PowerSupplyControl(plugs.BasePlug):   # pylint: disable=no-init
+  """Acronmae power module control plug."""
+  
+
+  def __init__(self):
+    config = conf.Config()
+    self.serial_number = config.serial_number_of_EtherStem
+    self.voltage = config.default_power_module_voltage_output
+    self.current_limit = config.default_power_module_current_limit
+    self.MTMPowerModuleAddress = 6
+    self.routerAddress = 4
+    self.EtherStem = MTMEtherStem()
+    self.power_module = MTMPM1()
+
+  def DiscoverAndConnectModule(self):
+    """
+    Discover all EtherStems on network.
+    Connect the one with the serial number specified in config yaml file.
+    The connection from server to EtherStem is Ethernet via TCPIP.
+    """
+    # codes below is to discover EtherStem on network
+    stem_list = discover.find_all(Spec.TCPIP)
+    if not stem_list:
+      _LOG.info('None EtherStem is found.')
+      raise NoneEtherStemFoundError
+    for stem in stem_list:
+      # print "stem: "+str(stem)
+      if stem.serial_number == self.serial_number:
+        stem_index = stem_list.index(stem)
+        spec = stem_list[stem_index]
+        EtherStemFound = True
+        break
+    else:
+      _LOG.info('No Matching EtherStem is found.')
+      raise NoMatchingEtherStemFoundError
+      
+    
+    # codes below is to connect EtherStem
+    _LOG.info("Connecting to EtherStem with serial number: ",hex(self.serial_number).upper())
+    # print("Connecting to EtherStem with serial number: 0x%X" % self.serial_number)
+    # print "spec = "+str(spec)
+    res = self.EtherStem.connect_from_spec(spec)
+    # print("Result of Connecting to EtherStem: " +str(res))
+    if res != 0:
+      _LOG.info("Error of connecting to EtherStem:"+str(res))
+      raise ConnectingToEtherStemError
+    else:
+      _LOG.info("Connecting to EtherStem Succeed.")
+
+    # codes below is to connect Power Module through EtherStem
+
+    res = self.EtherStem.i2c[0].setPullup(1)
+    CheckReturnCode(res, "setPullUp")
```

Comment:
I would consider wrapping these API calls in a friendlier interface that does API return code checking for you rather than having a separate helper function for checking return codes.